torch/_inductor/fx_passes/split_cat.py

```diff
@@ -122,7 +122,15 @@ def normalize_cat_default(match: Match, *args, **kwargs):
         return
 
     ndim = tensors[0].meta["example_value"].dim()
-    assert all(ndim == x.meta["example_value"].dim() for x in tensors)
+
+    def is_empty_tensor(x):
+        # special case where torch.cat supports cat'ing with an empty tensor
+        x_shape = x.meta["example_value"].shape
+        return len(x_shape) == 1 and x_shape[0] == 0
+
+    assert all(
```

Comment:
curious in general for special case like this, do we filter out empty tensors in the fx pass directly (i.e. remove that node from `tensors`), or we do this when creating inductor IR?

Reply:
good q. i only intended to do the change to ir.py, but the test i added failed on the pass.  i am not sure if there is a better way to handle this.